Buddies/Utilities/AlgoliaSearch.swift

```diff
@@ -0,0 +1,139 @@
+//
+//  AlgoliaSearch.swift
+//  Buddies
+//
+//  Created by Noah Allen on 2/12/19.
+//  Copyright © 2019 Jack and the Beans. All rights reserved.
+//
+
+/** Usage example:
+let search = AlgoliaSearch()
+let s = "2019/02/15".toDate()?.date
+let e = "2019/02/20".toDate()?.date
+search.searchActivities(startingAt: s, endingAt: e, atLocation: (41.1603988, -80.0866907)) { (activities: [String], err: Error?) in
+ 
+    if let error = err { print(error) }
+    print(activities)
+}
+**/
+
+import Foundation
+import InstantSearchClient
+
+protocol SearchIndex {
+    func search(_ query: Query, requestOptions: RequestOptions?, completionHandler: @escaping CompletionHandler) -> Operation
+}
+extension Index: SearchIndex {}
+
+protocol SearchClient {
+    func getIndex(withName indexName: String) -> SearchIndex
+}
+
+// Wrap the function so that we can use "SearchIndex" as a mock
+extension Client: SearchClient {
+    func getIndex(withName indexName: String) -> SearchIndex {
+        return index(withName: indexName)
+    }
+}
+
+class AlgoliaSearch {
+    static let ACTIVITY_INDEX = "BUD_ACTIVITIES"
+    var client: SearchClient
+
+    init (algoliaClient: SearchClient? = nil) {
+        self.client = algoliaClient != nil ? algoliaClient! : AlgoliaSearch.algoliaFactory()
```

Comment:
This could be `algoliaClient ?? AlgoliaSearch.algoliaFactory()` which I think looks a little nicer.

Not crucial.

Reply:
Good call